pkg/apis/planetscale/v2/vitessshard_types.go

```diff
@@ -134,6 +134,10 @@ type VitessReplicationSpec struct {
 	//
 	// Default: Semi-sync is not enforced.
 	EnforceSemiSync bool `json:"enforceSemiSync,omitempty"`
+
+	// InitializeMaster can be used to enable or disable the operator initializing replication
+	// on a new or restored Vitess Shard object.
+	InitializeMaster bool `json:"initializeMaster,omitempty"`
```

Comment:
I believe you still will want to copy the setting on down with the following logic:
1. If child has setting (non-nil), use child setting, 
2. else copy parent down.

this allows someone to specify the setting at the cluster level and have overrides at the shard level. In other words, I could turn it off globally, and then turn it on for select keyspaces, or select shards.

Reply:
We don't have a cluster-level Replication field in VitessCluster yet. For now, you have to specify this individually for every shard template.

It would make sense to add a cluster-level field in the future, but the override logic will live in the controller that generates VitessShard objects, not in defaulting code. Defaulting code needs to avoid taking values from one place and sticking them somewhere else; we learned from mistakes in core k8s that that breaks compatibility.

Reply:
I put the comment here because I wasn't talking about defaulting code.  Sorry if my comment was unclear.